### PR TITLE
Add getSupportedDocuments API to Barbershop

### PR DESCRIPTION
**Pull request by wesleyk**
This enables clients to determine what types of Documents
are supported for a given DocumentData.

### CODE REVIEW

File: barber/src/main/kotlin/app/cash/barber/Barbershop.kt [hunk 1]
```diff
@@ -15,6 +15,9 @@ interface Barbershop {
     documentClass: KClass<out D>
   ): Barber<DD, D>
 
+  fun <DD : DocumentData> getSupportedDocuments(documentDataClass: KClass<out DD>):
```
Comment (adrw) [inline barber/src/main/kotlin/app/cash/barber/Barbershop.kt]: 1) I lean towards `getTargetDocuments` as the function name to match the use of `target` in the DocumentTemplate API. Thoughts on changing the function name to that?

2) Could you add a similar inline function with reified type to the bottom of this file similar to what exists for `getBarber`? Providing that and using it in tests would match the idiomatic style present throughout the rest of the library.

File: barber/src/main/kotlin/app/cash/barber/RealBarbershop.kt [hunk 2]
```diff
@@ -44,6 +44,18 @@ internal class RealBarbershop(
...
+    documentDataClass: KClass<out DD>
+  ): Set<KClass<out Document>> {
+    val supportedDocuments = mutableSetOf<KClass<out Document>>()
+    barbers.keys.forEach {
+      if (it.documentData.equals(documentDataClass)) {
+        supportedDocuments.add(it.document)
+      }
+    }
+    return supportedDocuments.toSet()
+  }
```
Comment (hamdanjaveed) [inline barber/src/main/kotlin/app/cash/barber/RealBarbershop.kt]: nit: could be written as

```
override fun <DD : DocumentData> getTargetDocuments(
  documentDataClass: KClass<out DD>
): Set<KClass<out Document>> = barbers.keys.filter { it.documentData == documentDataClass }
```

Comment (hamdanjaveed) [inline barber/src/main/kotlin/app/cash/barber/RealBarbershop.kt]: small nit: could be written as

```
override fun <DD : DocumentData> getTargetDocuments(documentDataClass: KClass<out DD>): Set<KClass<out Document>> = barbers.keys
  .filter { it.documentData == documentDataClass }
  .map { it.document }
  .toSet()
```

**Comment by adrw** [comment]
Looks good!

**Pull request merged**